### ISSUE TITLE
Add in error attempting to follow this tutorial

### ISSUE THREAD

**Issue by devhawk** [OfficeDev/office-js-docs-pr#2937]
![image](https://user-images.githubusercontent.com/8965/133002884-e0f4a82c-1629-461a-b4cd-df61540794b8.png)

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 9c4ddde7-f1fd-44ba-fba1-0ae9d0131df7
* Version Independent ID: d946ffe6-f29f-1563-b679-121090ba60a1
* Content: [Build your first Excel task pane add-in - Office Add-ins](https://docs.microsoft.com/en-us/office/dev/add-ins/quickstarts/excel-quickstart-jquery?tabs=yeomangenerator)
* Content Source: [docs/quickstarts/excel-quickstart-jquery.md](https://github.com/OfficeDev/office-js-docs-pr/blob/master/docs/quickstarts/excel-quickstart-jquery.md)
* Product: **excel**
* Technology: **add-ins**
* GitHub Login: @o365devx
* Microsoft Alias: **o365devx**

**Comment by devhawk** [comment]
This appears to be the same issue as #1598. I tried deleting the folder `C:\Users\{username}\.office-addin-dev-certs` then running `npx office-addin-dev-certs install` but it did not solve the problem

**Comment by AlexJerabek** [comment]
Hi @devhawk,

Thanks for reporting this issue. Let's see if we can help unblock you. I have a couple of questions: 

- What platform are you running on (Windows, Mac, Chrome, Edge, etc)?
- What version of yo office are you using (`yo office --version`)?
- When you first ran `npm start`, did a message pop-up asking you to install a certificate?
- Have you tried [clearing the Office cache](https://docs.microsoft.com/en-us/office/dev/add-ins/testing/clear-cache)? That often fixes problems with your add-in or Office being in a weird state.

**Comment by devhawk** [comment]
* Platform: Windows 10 version 20H2 (OS Build 19042.1165)
* yo office version: 4.3.0
* I didn't see a windows style popup, but I recall a certificate install both originally and when I deleted `.office-addin-dev-certs` folder and ran `npx office-addin-dev-certs install`
* I had not tried that, but I just did and it made no difference

**Comment by AlexJerabek** [comment]
Thanks @devhawk.

@millerds, could you please help investigate?

**Comment by millerds** [comment]
Can you run "npx office-addin-dev-certs uninstall" a few times to make sure old certificates are removed before running install.

**Comment by devhawk** [comment]
I ran `npx office-addin-dev-certs uninstall` several times. That command appears to have removed `~/.office-addin-dev-certs` directory the first time I ran it, but I ran it "a few times" as per instruction.

I then followed the [Excel Add-in Quickstart instructions](https://docs.microsoft.com/en-us/office/dev/add-ins/quickstarts/excel-quickstart-jquery?tabs=yeomangenerator). I didn't get a cert install popup, but the node terminal window that popped up after I ran `npm start` from my add-in directory started with the following text:

```
Uninstalling CA certificate "Developer CA for Microsoft Office Add-ins"...
You no longer have trusted access to https://localhost.
The developer certificates have been generated in C:\Users\harry\.office-addin-dev-certs
Installing CA certificate "Developer CA for Microsoft Office Add-ins"...
You now have trusted access to https://localhost.
Certificate: C:\Users\harry\.office-addin-dev-certs\localhost.crt
Key: C:\Users\harry\.office-addin-dev-certs\localhost.key
```

Once Excel launched, I clicked on the "Show Taskpane" button in the Home ribbon menu. I got a dialog box explaining how to attach the VSCode debugger, and clicked "OK" to keep getting the dialog in the future. The task panel titled "My Office Add-in" appears, but its contents was the same "ADD-IN ERROR" window I included in the original issue above.

**Comment by millerds** [comment]
What version of Office are you running?

**Comment by devhawk** [comment]
Whatever is installed with Office 365. From the about box: Microsoft® Excel® for Microsoft 365 MSO (16.0.14326.20164) 64-bit

**Comment by devhawk** [comment]
Note, I'm having this issue w/ two separate machines.

**Comment by devhawk** [comment]
My second machine is running Microsoft® Excel® for Microsoft 365 MSO (16.0.14326.20324) 64-bit and is having the same issue. I followed the same process I [described above](https://github.com/OfficeDev/office-js-docs-pr/issues/2937#issuecomment-919487518) and got the same result

**Comment by millerds** [comment]
It appears that the cert that is being found is an invalid one and so a new one is being installed, but that doesn't seem to help.  I'm trying to determine where a rogue bad cert might be coming from.  Can you double check that the certs that get added to .office-addin-dev-certs have valid date they can be used from?

**Comment by millerds** [comment]
It is expected that they can't be verified because we are generating and using them locally.  You should only allow certs like this if you trust the person giving them to you (thanks for your trust by the way :-) ).

Next I'd like to have you run some powershell script to see if there is an error happening under the covers of our tool.  Open a powershell prompt and run 'Get-Command -name Import-Certificate' and if that works please run 'Import-Certificate -CertStoreLocation cert:\$machine\Root ${caCertificatePath}'  where caCertificatePath is the path to the ca.crt file.

Let us know if there are any errors when trying to do this.

**Comment by devhawk** [comment]
running `Import-Certificate` appears to have solved the issue. I get the black circle with "Logo" in it on the task pane and more importantly the task pane opens without error!

**Comment by millerds** [comment]
Hmm . . . that is what the office-addin-dev-certs command is supposed to be doing.  You didn't encounter any problematic messages when running the powershell?

**Comment by devhawk** [comment]
> Hmm . . . that is what the office-addin-dev-certs command is supposed to be doing. You didn't encounter any problematic messages when running the powershell?

Not that I recall. I was having this issue on two separate machines but I only ran `Import-Certificate` on one of them. I will try on my other machine tomorrow

**Comment by devhawk** [comment]
On my second machine, I ran `npx office-addin-dev-certs uninstall`, then regenerated the add in via the instructions at https://docs.microsoft.com/en-us/office/dev/add-ins/quickstarts/excel-quickstart-jquery?tabs=yeomangenerator and finally invoked `npm start` from the `My Office Add-in` folder.

Main terminal window output:
```
$ npm start

> office-addin-taskpane-js@0.0.1 start
> office-addin-debugging start manifest.xml

Debugging is being started...
App type: desktop
Enabled debugging for add-in 346d7e59-a312-46aa-8a0e-e708779f55aa.
Starting the dev server... (webpack-dev-server --mode development)
The dev server is running on port 3000. Process id: 21080
Sideloading the Office Add-in...
Debugging started.
```

Launched terminal window output:
```
Uninstalling CA certificate "Developer CA for Microsoft Office Add-ins"...
You no longer have trusted access to https://localhost.
The developer certificates have been generated in C:\Users\harry\.office-addin-dev-certs
Installing CA certificate "Developer CA for Microsoft Office Add-ins"...
You now have trusted access to https://localhost.
Certificate: C:\Users\harry\.office-addin-dev-certs\localhost.crt
Key: C:\Users\harry\.office-addin-dev-certs\localhost.key
i ｢wds｣: Project is running at https://localhost:3000/
i ｢wds｣: webpack output is served from /
i ｢wds｣: Content not from webpack is served from C:\Users\harry\Source\playground\My Office Add-in
i ｢wdm｣: Hash: 9c66078e853ceb304e7f
Version: webpack 4.46.0
Time: 7532ms
Built at: 09/24/2021 10:38:41 AM
                 Asset       Size    Chunks                   Chunk Names
    assets/icon-16.png   1.56 KiB            [emitted]
    assets/icon-32.png   2.33 KiB            [emitted]
    assets/icon-80.png   4.72 KiB            [emitted]
assets/logo-filled.png   11.6 KiB            [emitted]
         commands.html  473 bytes            [emitted]
           commands.js    356 KiB  commands  [emitted]        commands
       commands.js.map    408 KiB  commands  [emitted] [dev]  commands
      manifest.dev.xml   4.09 KiB            [emitted]
           polyfill.js    991 KiB  polyfill  [emitted]        polyfill
       polyfill.js.map    944 KiB  polyfill  [emitted] [dev]  polyfill
          taskpane.css   1.48 KiB            [emitted]
         taskpane.html    2.5 KiB            [emitted]
           taskpane.js    360 KiB  taskpane  [emitted]        taskpane
       taskpane.js.map    408 KiB  taskpane  [emitted] [dev]  taskpane
Entrypoint polyfill = polyfill.js polyfill.js.map
Entrypoint taskpane = taskpane.js taskpane.js.map
Entrypoint commands = commands.js commands.js.map
[0] multi (webpack)-dev-server/client?https://localhost:3000 core-js/stable regenerator-runtime/runtime 52 bytes {polyfill} [built]
[2] multi (webpack)-dev-server/client?https://localhost:3000 ./src/taskpane/taskpane.js 40 bytes {taskpane} [built]
[3] multi (webpack)-dev-server/client?https://localhost:3000 ./src/commands/commands.js 40 bytes {commands} [built]
[./assets/icon-16.png] 64 bytes {taskpane} [built]
[./assets/icon-32.png] 64 bytes {taskpane} [built]
[./assets/icon-80.png] 64 bytes {taskpane} [built]
[./node_modules/core-js/es/index.js] 8.89 KiB {polyfill} [built]
[./node_modules/core-js/internals/path.js] 71 bytes {polyfill} [built]
[./node_modules/core-js/stable/index.js] 102 bytes {polyfill} [built]
[./node_modules/core-js/web/index.js] 398 bytes {polyfill} [built]
[./node_modules/regenerator-runtime/runtime.js] 24 KiB {polyfill} [built]
[./node_modules/strip-ansi/index.js] 161 bytes {polyfill} {taskpane} {commands} [built]
[./node_modules/webpack-dev-server/client/index.js?https://localhost:3000] (webpack)-dev-server/client?https://localhost:3000 4.29 KiB {polyfill} {taskpane} {commands} [built]
[./src/commands/commands.js] 1.06 KiB {commands} [built]
[./src/taskpane/taskpane.js] 3.24 KiB {taskpane} [built]
    + 409 hidden modules
Child HtmlWebpackCompiler:
                     Asset      Size  Chunks             Chunk Names
    assets/logo-filled.png  11.6 KiB          [emitted]
     + 2 hidden assets
    Entrypoint HtmlWebpackPlugin_0 = __child-HtmlWebpackPlugin_0
    Entrypoint HtmlWebpackPlugin_1 = __child-HtmlWebpackPlugin_1
    [./assets/logo-filled.png] 68 bytes {HtmlWebpackPlugin_0} [built]
    [./node_modules/html-webpack-plugin/lib/loader.js!./src/commands/commands.html] 450 bytes {HtmlWebpackPlugin_1} [built]
    [./node_modules/html-webpack-plugin/lib/loader.js!./src/taskpane/taskpane.html] 2.61 KiB {HtmlWebpackPlugin_0} [built]
i ｢wdm｣: Compiled successfully.
```

On this machine (where I have NOT run `Import-Certificate` from PS) I get the Add-in error

**Comment by millerds** [comment]
This looks like correct output, but clearly the cert installing script is silently failing an reporting a success.  Given that you can run it manually via PS it looks like it should work, but something about running the PS via the child_process package using a 'powershell -ExecutionPolicy Bypass -File "${script"}' command (where '${script}' the the path to a .ps1 file) isn't reporting an error.

Since I don't have a repro it's not something I can investigate . . . but you do have a workaround that you shouldn't have to repeated do.

**Comment by devhawk** [comment]
Yup, I have a workaround

**Comment by AlexJerabek** [comment]
Hi @devhawk,
Since you have a workaround for this machine and a fix for the other, can we close this issue?

**Comment by arnoudcommandeur** [comment]
Hi @devhawk, actually I'm facing the same problem. I've tried all the steps above, but still have te same problem. I'm trying the example project at https://github.com/OfficeDev/PnP-OfficeAddins/tree/main/Samples/outlook-set-signature. I get the error when I press the add-in button to show the taskpane of the add-in when I compose a new Message within Outlook.

I have tried to run the example from localhost (see section: Run the sample from localhost) and I also get the same error.

What I've already tried:
* Uninstalled the certificates
* Recreated the certificates
* Added the newly created localhost.crt certificate by using the Powershell script

I get the same error:
![image](https://user-images.githubusercontent.com/2140759/143139111-a4d29eef-58b9-4217-88d1-4ce3d0750ae3.png)

Any help is welcome!

Windows 10
Outlook version 16.0.14718.200000 / 64 bits

**Comment by AlexJerabek** [comment]
Hi @arnoudcommandeur,
Could you plese raise this as a new issue in the [generator-office repo](https://github.com/OfficeDev/generator-office/issues)? That would be a good place to flag general issues related to yo office and the certificates it creates.
Thanks!

**Comment by ANHPearce** [comment]
@AlexJerabek 

I added this to the generator-office repo as you requested. I had run into the same error. 

Solved it by manually installed the certs that had been generated. See my comments on the other repo.
